docs/sections/contributor_guide/developer_setup.rst

```diff
@@ -1,5 +1,147 @@
+.. include:: ../links.rst
+
 ***************
 Developer Setup
 ***************
 
-TBD
+Creating a ``bash`` development shell
+=====================================
+
+
+This recipe uses the ``aarch64`` (64-bit ARM) Miniforge for Linux, and installs into ``$HOME/conda``. Adjust as necessary for your target system.
+
+1. Download, install, and activate the latest `Miniforge3`_ for your system. If an existing conda (Miniforge, Miniconda, Anaconda, etc.) installation is available and writable, you may activate that and skip this step and continue on to the next.
+
+   .. code:: sh
+   
+     wget https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-aarch64.sh
+     bash Miniforge3-Linux-aarch64.sh -bfp ~/conda
+     rm Miniforge3-Linux-aarch64.sh
+     source ~/conda/etc/profile.d/conda.sh
+     conda activate
+
+2. Install the `condev`_ package into the base environment.
+
+   .. code:: sh
+
+     conda install -y -c maddenp condev
+
+3. In a clone of the `workflow-tools repository`_, create the development shell.
+
+
+   .. code:: sh
+
+     cd /to/your/workflow-tools/clone
+     make devshell
+
+
+.. _paragraph: 
+
+If the above is successful, you will be in a ``workflow-tools`` development shell. See below for usage information. You may exit the shell with ``exit`` or ``ctrl-d``.
+
+Future ``make devshell`` invocations will be almost instantaneous, as the underlying virtual environment will already exist. In general, all source code changes will be immediately live in the development shell, subject to execution, test, etc. But some changes – especially to the contents of the ``recipe/`` directory, or to the ``src/setup.py`` module – may require recreation of the development shell. If you know this is needed, or when in doubt: 
+
+    1. Exit the development shell, run ``conda env remove -n DEV-uwtools`` to remove the old environment.
+    2. Run ``make devshell`` to recreate it.
+
+If your development shell misses any functionality you’re used to in your main (``bash``) shell, you can create a ``~/.condevrc`` file, which will be sourced by ``make devshell``. When in doubt, you might:
+
+   .. code::
+
+     cat <<EOF >~/.condevrc
+     source ~/.bashrc
+     EOF
+
+
+Using a ``bash`` development shell
+==================================
+
+In an active development shell, the following ``make`` targets are available and act on all ``.py`` files under ``src/``:
+
+Formats Python code with black, imports with isort, docstrings with docformatter, and .jsonschema documents with jq.
```

Comment:
Do we need this line? It seems to repeat what's in the table below.

Reply:
Yes, this was left in from my edits. I'll remove this shortly.